src/software/jetson_nano/services/motor.cpp

```diff
@@ -210,75 +205,146 @@ bool MotorService::checkDriverFault(uint8_t motor)
+        motor_faults.insert(TbotsProto::MotorFault::PHASE_U_SHORT_TO_VS_DETECTED);
+        drive_enabled = false;
     }
 
     if (gstat_bitset[8])
     {
         LOG(WARNING) << "shortdet_v: V short counter has triggered at least once.";
+        motor_faults.insert(TbotsProto::MotorFault::PHASE_V_SHORT_COUNTER_DETECTED);
     }
 
     if (gstat_bitset[9])
     {
         LOG(WARNING) << "s2gv: Short to GND detected on phase V."
                      << "The driver becomes disabled until flag becomes cleared.";
+        motor_faults.insert(TbotsProto::MotorFault::PHASE_V_SHORT_TO_GND_DETECTED);
+        drive_enabled = false;
     }
 
     if (gstat_bitset[10])
     {
         LOG(WARNING) << "s2vsv: Short to VS detected on phase V."
                      << "The driver becomes disabled until flag becomes cleared.";
+        motor_faults.insert(TbotsProto::MotorFault::PHASE_V_SHORT_TO_VS_DETECTED);
+        drive_enabled = false;
     }
 
     if (gstat_bitset[12])
     {
         LOG(WARNING) << "shortdet_w: short counter has triggered at least once.";
+        motor_faults.insert(TbotsProto::MotorFault::PHASE_W_SHORT_COUNTER_DETECTED);
     }
 
     if (gstat_bitset[13])
     {
         LOG(WARNING) << "s2gw: Short to GND detected on phase W."
                      << "The driver becomes disabled until flag becomes cleared.";
+        motor_faults.insert(TbotsProto::MotorFault::PHASE_W_SHORT_TO_GND_DETECTED);
+        drive_enabled = false;
     }
 
     if (gstat_bitset[14])
     {
         LOG(WARNING) << "s2vsw: Short to VS detected on phase W."
                      << "The driver becomes disabled until flag becomes cleared.";
+        motor_faults.insert(TbotsProto::MotorFault::PHASE_W_SHORT_TO_VS_DETECTED);
+        drive_enabled = false;
     }
 
-    return !gstat_bitset.any();
+    return MotorFaultIndicator(drive_enabled, motor_faults);
+}
+
+TbotsProto::MotorStatus MotorService::updateMotorStatus(double front_left_velocity_mps,
+                                                        double front_right_velocity_mps,
+                                                        double back_left_velocity_mps,
+                                                        double back_right_velocity_mps)
+{
+    TbotsProto::MotorStatus motor_status;
+
+    cached_motor_faults_[motor_fault_detector] = checkDriverFault(motor_fault_detector);
+
+    for (uint8_t i = 0; i < NUM_DRIVE_MOTORS; ++i)
+    {
+        TbotsProto::DriveUnit drive_status;
+        drive_status.set_drive_enabled(
+            cached_motor_faults_[motor_fault_detector].drive_enabled);
+
+        for (const TbotsProto::MotorFault& fault :
+             cached_motor_faults_[motor_fault_detector].motor_faults)
+        {
+            drive_status.add_motor_fault(fault);
+        }
+
+        if (motor_fault_detector == FRONT_LEFT_MOTOR_CHIP_SELECT)
+        {
+            drive_status.set_wheel_velocity(static_cast<float>(front_left_velocity_mps));
```

Comment:
nit: Since we have the up to date wheel velocities for all wheels, I think this line should be moved out of the if statement so the proto always has the up to date velocity for all wheels.